### PR TITLE
Backbone Webdav adapter MKCOL support

### PR DESCRIPTION
**Pull request by PVince81**
## Description
Backbone Webdav Adapter MKCOL support
    
Usually Backbone collections cannot be created and just simply exists.
But in the Webdav world they need to be creatable.
    
This enhancement makes it possible to use a Backbone Model to represent such collections and when creating it, it will use MKCOL instead of PUT.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by PVince81** [comment]
Please review @DeepDiver1975 @mjobst-necls @jvillafanez

File: core/js/oc-backbone-webdav.js [hunk 1]
```diff
@@ -120,7 +120,12 @@
 		}
 
 		var parts = url.split('/');
-		return parts[parts.length - 1];
+		var result;
+		while (!result && parts.length > 0) {
```
Comment (jvillafanez) [inline core/js/oc-backbone-webdav.js]: A "do-while" loop might be better, and also a comment for the special case that needs this.

File: core/js/oc-backbone-webdav.js [hunk 2]
```diff
@@ -259,9 +285,14 @@
 		var params = {type: methodMap[method] || method};
 		var isCollection = (model instanceof Backbone.Collection);
 
-		if (method === 'update' && (model.usePUT || (model.collection && model.collection.usePUT))) {
-			// use PUT instead of PROPPATCH
-			params.type = 'PUT';
+		if (method === 'update') {
+			if (model.getInnerCollection) {
```
Comment (jvillafanez) [inline core/js/oc-backbone-webdav.js]: The naming is weird. If `getInnerCollection` is a function the normal check should be calling the function and verify the inner collection is there. If it's a property maybe we should change the name, or at least comment that this is a property and not a function.

Comment (PVince81) [inline core/js/oc-backbone-webdav.js]: This line checks whether the method exists to find out whether this model has an inner collection. Maybe that looks too hacky. I'll try and think of a more obvious way.

Comment (jvillafanez) [inline core/js/oc-backbone-webdav.js]: A comment might be enough as long as it makes clear this is intended and the reasons to do this way.

Comment (PVince81) [inline core/js/oc-backbone-webdav.js]: Ok, understood. I'll add a comment and replace with `_.isFunction(model.getInnerCollection)` which is also much clearer about its intent.

**Comment by PVince81** [comment]
@jvillafanez all adjusted. Please re-review, thanks.

**Comment by jvillafanez** [comment]
👍

**Comment by PVince81** [comment]
Do I get an achievement for "weirdest random failure output" on Jenkins ?
```
org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
General error during canonicalization: java.lang.NoClassDefFoundError: org/jenkinsci/plugins/pipeline/modeldefinition/ast/ModelASTJobProperties

java.lang.RuntimeException: java.lang.NoClassDefFoundError: org/jenkinsci/plugins/pipeline/modeldefinition/ast/ModelASTJobProperties
	at org.codehaus.groovy.control.CompilationUnit.convertUncaughtExceptionToCompilationError(CompilationUnit.java:1089)
```

**Pull request merged**